### PR TITLE
add special objects

### PR DESCRIPTION
**Pull request by alexander1996**

### CODE REVIEW

File: svg/specialObjects.go [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+	style := fmt.Sprintf("stroke:black; stroke-width: %v; fill: none; stroke-dasharray: 10;", weight)
+	styleArrow := fmt.Sprintf("stroke:black; stroke-width: %v; fill: none;", weight)
+	transformation := fmt.Sprintf("rotate(%v,%v,%v)", route.rotateAngel, route.centerX, route.centerY)
+
+	canvas.Group("id=\"id" + strconv.Itoa(object.Id) + "\"")
+	canvas.CSS(prefixSelectors(object.CSS, object.Id))
+	canvas.Gtransform(transformation)
+	canvas.Line(route.rightLinePointX, route.centerY, route.leftLinePointX, route.centerY, style)
+	canvas.Polyline(route.arrowXs, route.arrowYs, styleArrow)
+	canvas.Gend()
```
Comment (daynin) [inline svg/specialObjects.go]: Why do you call Gend() method 2 times here?

File: svg/specialObjects.go [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+		0, false, true,
+		area.rightLinePointX,
+		area.rightLinePointY+int(2*area.radiusY))
+	canvas.Arc(area.leftLinePointX,
+		area.leftLinePointY,
+		area.radiusX,
+		area.radiusY,
+		0, false, true,
+		area.leftLinePointX,
+		area.leftLinePointY-int(2*area.radiusY))
```
Comment (daynin) [inline svg/specialObjects.go]: Why do you call Gend() method 2 times here?

File: svg/svg.go [hunk 3]
```diff
@@ -190,5 +191,50 @@ func RenderTile(tile *mapobjects.Tile, objects *[]mapobjects.MapObject, writer i
...
+func renderSpecialObject(canvas *svg.SVG, tile *mapobjects.Tile) {
+	patrollingArea, _ := mapobjects.NewObject(
+		32,
+		"LINESTRING (70.6 50.6, 16.183333 51.716667)",
+		`polyline, path, line {
+	           stroke:black; 
+			   stroke-width:1;
+			   fill: none
+	         }`)
+
```
Comment (daynin) [inline svg/svg.go]: Your methods can return an error, why you don't check it? Maybe this is the error that crash whole app

File: svg/specialObjects.go [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+import (
+	"fmt"
+	"math"
+	"strconv"
+
+	"github.com/TerraFactory/svgo"
+	"github.com/TerraFactory/tilegenerator/mapobjects"
+	"github.com/paulsmith/gogeos/geos"
+)
+
```
Comment (daynin) [inline svg/specialObjects.go]: All public names should be uppercase

File: svg/specialObjects.go [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+		xs = append(xs, centerX+int(point.Y*beamDiagram.radius/maxValue))
+		ys = append(ys, centerY+int(point.X*beamDiagram.radius/maxValue))
+	}
+
+	xs[0] = centerX + int(beamDiagram.radius)
+	ys[0] = centerY
+
+	return xs, ys
+}
+
```
Comment (daynin) [inline svg/specialObjects.go]: @alexander1996 , please, add tests for functions like this one (math functions)

File: svg/specialObjects.go (hunk 4 above)
Comment (alexander1996) [inline svg/specialObjects.go]: chartPoint is private type, which use only for calculation

File: svg/specialObjects.go (hunk 1 above)
Comment (alexander1996) [inline svg/specialObjects.go]: Methods canvas.Group and canvas.Gtransform are used in pair with canvas.Gend.

File: svg/specialObjects.go (hunk 4 above)
Comment (daynin) [inline svg/specialObjects.go]: @alexander1996 

Ok, but what about fields of chartPoint? Are they private too?

File: svg/specialObjects.go (hunk 1 above)
Comment (daynin) [inline svg/specialObjects.go]: Did your try call canvas.Gend() just one time? What happens? I think, it'll be enough to commit changes on the canvas.

**Pull request merged**